### DISCUSSION TITLE
关于第一个exercise的问题

### DISCUSSION THREAD

**Discussion by LucienXian** [mlc-ai/mlc-zh#37] [Q&A]
在上周的课程中提到了空间轴和规约轴，这两个概念对应到tvm的使用时会有什么区别吗（这里面对于空间轴和规约轴的运行会产生一些区别？），假如混用会有什么影响吗？
我在做二维卷积的时候我发现下面两种方式都能正常通过测试，即输入通道q对应到空间轴也没问题（这是因为q的范围内只有一个元素？）
vb, vk, vi, vj, vq, vdi, vdj = T.axis.remap("SSSSRRR", [b, k, i, j, q, di, dj])
vb, vk, vi, vj, vq, vdi, vdj = T.axis.remap("SSSSSRR", [b, k, i, j, q, di, dj])

**Comment by Hzfengsy** [comment]
感谢提问！关于不同的axis_type我们分两个情况讨论：
1. 在编译运行时，所有的循环都会被lower成普通的for循环（如果在CPU上）所以即使标错也不会影响正确性；
2. 在变换(transform)时，会检查axis_type，如果标错可能导致变换后程序错误，或者本应合法的变换无法进行

**Reply by LucienXian** [reply]
> 

vb, vk, vi, vj, vq, vdi, vdj = T.axis.remap("SSSSSSS", [b, k, i, j, q, di, dj])

但这样似乎会有问题（这里没有transform？）

**Reply by Hzfengsy** [reply]
可能是因为block本身含有init部分，则block要求是一个reduction block（需要至少一个reduce_axis）
总的来说还是需要用户手动指定正确的axis_type，否则出现的错误可能是为定义行为
